rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+            },
+        );
+
+        Ok((new_current_ans_lookups, new_ans_lookups))
+    }
+
+    pub fn from_primary_name_record_table_delete_table_item_v1(
+        delete_table_item: &DeleteTableItem,
+        txn_version: i64,
+        txn_timestamp: chrono::NaiveDateTime,
+        address_to_primary_name_record: &AddressToPrimaryNameRecord,
+        conn: &mut PgPoolConnection,
+    ) -> anyhow::Result<(
+        HashMap<CurrentAnsLookupPK, Self>,
+        HashMap<AnsLookupPK, AnsLookup>,
+    )> {
+        let mut new_current_ans_lookups = HashMap::new();
+        let mut new_ans_lookups = HashMap::new();
+
+        let table_item_data = delete_table_item.data.as_ref().unwrap();
+        let key_type = table_item_data.key_type.as_str();
+        let key = serde_json::from_str(&table_item_data.key).unwrap_or_else(|e| {
+            panic!(
+                "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                txn_version, key_type, table_item_data.key, e
+            )
+        });
+        let registered_address = standardize_address(key);
+
+        // Need to lookup previous name record to update the row's is_primary = false
+        let mut previous_name_record = match Self::lookup_name_record_by_primary_address(
+            address_to_primary_name_record,
+            &registered_address,
+            conn,
+        ) {
+            Some(ans_record) => ans_record,
+            None => {
+                tracing::error!(
+                        transaction_version = txn_version,
+                        lookup_key = &registered_address,
+                        "Failed to get ans record for registered address. You probably should backfill db."
+                    );
+                return Ok((HashMap::new(), HashMap::new()));
+            },
+        };
+        // Update the lookup's name record with the new data and return
+        previous_name_record.is_primary = false;
+        previous_name_record.last_transaction_version = txn_version;
+        new_current_ans_lookups.insert(
+            (
+                previous_name_record.domain.clone(),
+                previous_name_record.subdomain.clone(),
+            ),
+            previous_name_record.clone(),
+        );
+        new_ans_lookups.insert(
+            (
+                txn_version,
+                previous_name_record.domain.clone(),
+                previous_name_record.subdomain.clone(),
+            ),
+            AnsLookup {
+                transaction_version: txn_version,
+                domain: previous_name_record.domain,
+                subdomain: previous_name_record.subdomain,
+                registered_address: previous_name_record.registered_address,
+                expiration_timestamp: previous_name_record.expiration_timestamp,
+                token_name: previous_name_record.token_name,
+                is_primary: false,
+                is_deleted: previous_name_record.is_deleted,
```

Comment:
why is this not just true?

Reply:
is_deleted represents whether the whole record got deleted. 

Consider the case where NR WTI and PN DTI exist in the same transaction. is_deleted should persist from the current record.